### PR TITLE
Add remaining tokens to design-tokens.json

### PR DESCRIPTION
**Pull request by hannajones**
This should generate all the tokens we currently have in our three design tokens files, with a few caveats:

* I'm not generating the `unset` tokens that will be removed as part of [Bug 1845630](https://bugzilla.mozilla.org/show_bug.cgi?id=1845630)
* Things aren't in quite the same order and we're not using the `@media not (prefers-contrast)` blocks as discussed earlier

### CODE REVIEW

File: toolkit/themes/shared/design-system/config.js
Comment (hannajones) [inline toolkit/themes/shared/design-system/config.js]: This is just a placeholder the real solution is in https://github.com/FirefoxUX/gecko-dev/pull/8

File: toolkit/themes/shared/design-system/design-tokens.json
Comment (hannajones) [inline toolkit/themes/shared/design-system/design-tokens.json:83]: We may want / need a better way to represent shadow down the line (I doubt this would work for mobile platforms...). Seems out of scope for now but I can file a subtask/follow up at least.

File: toolkit/themes/shared/design-system/design-tokens.json
Comment (hannajones) [inline toolkit/themes/shared/design-system/design-tokens.json:97]: We can't reference variables that aren't included in our JSON...but I'm pretty sure this will be moot once @mstriemer's button patches land: https://phabricator.services.mozilla.com/D188122#change-FJ7SeIB1CQgQ

File: toolkit/themes/shared/design-system/design-tokens.json
Comment (mstriemer) [inline toolkit/themes/shared/design-system/design-tokens.json]: I think the brand/platform colours can go here rather than assigning them to their own tokens

File: toolkit/themes/shared/design-system/build/css/tokens-brand.css
Comment (mstriemer) [inline toolkit/themes/shared/design-system/build/css/tokens-brand.css]: Do you mind adding these to the `TOKEN_SECTIONS` array so their sorted into the file?

It would be nice if we don't put anything in Unspecified. I left a comment on the test ticket to fail if we have tokens in here

**Review by mstriemer — CHANGES_REQUESTED** [review]
Just a couple small comments. I didn't check if all the tokens are there/correct but this definitely gets us closer and probably makes sense to do that when we're looking to merge to central

File: toolkit/themes/shared/design-system/README.md
Comment (TGiles) [inline toolkit/themes/shared/design-system/README.md]: This file isn't showing up when I run `mach storybook`. Is that expected?

File: toolkit/themes/shared/design-system/build/css/tokens-platform.css
Comment (TGiles) [inline toolkit/themes/shared/design-system/build/css/tokens-platform.css]: I don't think we have this token anymore, so it can be removed.

File: toolkit/themes/shared/design-system/build/css/tokens-platform.css
Comment (TGiles) [inline toolkit/themes/shared/design-system/build/css/tokens-platform.css]: Should these be in a "Font size" section like how we currently have them? https://searchfox.org/mozilla-central/source/toolkit/themes/shared/design-system/tokens-platform.css

File: toolkit/themes/shared/design-system/build/css/tokens-platform.css
Comment (TGiles) [inline toolkit/themes/shared/design-system/build/css/tokens-platform.css]: I don't think we have this line-height token anymore, we can remove this right @jsimplicio?

File: toolkit/themes/shared/design-system/build/css/tokens-platform.css
Comment (TGiles) [inline toolkit/themes/shared/design-system/build/css/tokens-platform.css]: @jsimplicio this line-height-small token should be removed too right? Just noticed that we're using it in central but we don't have a definition for it as far as I can tell. https://searchfox.org/mozilla-central/search?q=line-he&path=toolkit%2Fthemes%2Fshared%2Fdesign-system&case=false&regexp=false

File: toolkit/themes/shared/design-system/build/css/tokens-shared.css
Comment (TGiles) [inline toolkit/themes/shared/design-system/build/css/tokens-shared.css]: I don't think we have this token in central anymore (https://searchfox.org/mozilla-central/search?q=--spacing-&path=&case=false&regexp=false)

**Review by TGiles — CHANGES_REQUESTED** [review]
Just some minor things to fix. I think I'm missing some context on what tokens are missing compared to what we have in central. If that's the case, feel free to ignore my comments about tokens existing in this PR and not central 👍

File: toolkit/themes/shared/design-system/README.md
Comment (hannajones) [reply toolkit/themes/shared/design-system/README.md]: I actually have no idea why this is showing up at all, I'll delete it. I think this is auto generated by `style-dictionary` lol

File: toolkit/themes/shared/design-system/design-tokens.json
Comment (hannajones) [reply toolkit/themes/shared/design-system/design-tokens.json]: Ah I see what you mean, but I'm curious if there's another reason we're bothering to assign to `--brand-color-accent`, `--platform-color-accent`, etc. @jsimplicio do we want to keep those tokens or just re-assign `--color-accent-primary` and it's variants in the brand/platform sheets?

Either way I would rather make that change as a separate patch and keep this focused on just replicating what we have in the existing files to avoid any confusion/possible scope creep.

Comment (hannajones) [reply toolkit/themes/shared/design-system/design-tokens.json]: For now I just sorted them into the color comment blocks same as we have in central

**Comment by jsimplicio** [comment]
Thanks for working on this!

I've been pushing some fixes to naming inconsistencies or issues with the design tokens files today ahead of us preparing to land this JSON. A few things are in Nightly now and I'm actively waiting for review/working on a couple other patches that will once again modify the tokens files.

@hannajones would you rather hold off rebasing and merging this until all these changes are in?

File: toolkit/themes/shared/design-system/build/css/tokens-platform.css
Comment (jsimplicio) [reply toolkit/themes/shared/design-system/build/css/tokens-platform.css]: That's correct, no more line heights.

File: toolkit/themes/shared/design-system/build/css/tokens-platform.css
Comment (jsimplicio) [reply toolkit/themes/shared/design-system/build/css/tokens-platform.css]: That's correct.

File: toolkit/themes/shared/design-system/tokens-brand.css
Comment (hannajones) [inline toolkit/themes/shared/design-system/tokens-brand.css]: I left this out on purpose since it's getting moved/removed in phabricator.services.mozilla.com/D202767

File: toolkit/themes/shared/design-system/design-tokens.json
Comment (hannajones) [inline toolkit/themes/shared/design-system/design-tokens.json]: Do we need both of these or can this be cleaned up? (same thing happens with "active" and "hover" here - it seems like we shouldn't need to redefine this in forced colors mode)

File: toolkit/themes/shared/design-system/tokens-brand.css
Comment (TGiles) [inline toolkit/themes/shared/design-system/tokens-brand.css:14]: I don't have an answer to this, but I wonder what we should do about the existing comments like this. Maybe it's fine if we drop them, maybe it isn't.

**Review by TGiles — APPROVED** [review]
ship it

**Review by mstriemer — APPROVED** [review]

**Pull request merged**
